sdks/python/apache_beam/examples/snippets/snippets.py

```diff
@@ -1171,46 +1171,54 @@ def expand(self, pcoll):
         | beam.CombinePerKey(sum))
 # [END model_library_transforms_count]
 
-def examples_troubleshooting_metrics(renames):
+def examples_troubleshooting_metrics(number_list, output_path):
   """DebuggingWordCount example snippets."""
-  import re
-
   import apache_beam as beam
 
+  p = TestPipeline()  # Use TestPipeline for testing.
+
   # [START metrics_usage_example]
-  class FilterTextFn(beam.DoFn):
-    """A DoFn that filters for a specific key based on a regular expression."""
+  class FilterPairsFn(beam.DoFn):
+    """A DoFn that filters for pairs, and uses metrics.
 
-    def __init__(self, pattern):
-      self.pattern = pattern
+    Metrics are used to count number of pairs, and measure the distribution
+    of elements in the input PCollection."""
+
+    def __init__(self):
       # A custom metric can track values in your pipeline as it runs. Create
-      # custom metrics matched_words and unmatched_words.
-      self.matched_words = Metrics.counter(self.__class__, 'matched_words')
-      self.umatched_words = Metrics.counter(self.__class__, 'umatched_words')
+      # custom metrics pair_counter and collection_distribution
+      self.pair_counter = Metrics.counter(self.__class__, 'pair_counter')
+      self.collection_distribution = Metrics.distribution(
+          self.__class__, 'collection_distribution')
 
     def process(self, element):
-      word, _ = element
-      if re.match(self.pattern, word):
-        # Add 1 to the custom metric counter matched_words
-        self.matched_words.inc()
+      self.collection_distribution.update(element)
+      if element % 2 == 0:
+        self.pair_counter.inc()
         yield element
-      else:
-        # Add 1 to the custom metric counter umatched_words
-        self.umatched_words.inc()
 
-  p = TestPipeline()  # Use TestPipeline for testing.
-  filtered_words = (
+
+  filtered_numbers = (
```

Comment:
Move this up before the [START metrics_usage_example], just have the definition of the DoFn and `words | beam.ParDo(...)` here.